### ISSUE TITLE
Support for variable postcss plugins

### ISSUE THREAD

**Issue by bendyorke** [bendyorke/boot-postcss#1]

**Comment by stephenway** [comment]
I'd vote for including whatever combination of plugins you want. Until then this project doesn't meet my requirements.

**Comment by bendyorke** [comment]
@stephenway definitely!  This is the first feature I'm working on :)  I need to figure out how to pull in those plugins though.  Until I add support for running postcss with node, I think all the plugins would need to be packaged in here (or in a seperate repo) cljsjs style.  

If that's the case, what postcss plugins would be most important to you?

**Comment by stephenway** [comment]
@bendyorke autoprefixer, mixins, media, conditionals, selectors, custom-props, nested, selector-not, calc, extend, contrast, logical-props, flexbugs-fixes, mdcss, mqpacker, cssnano. However, ultimately I want control over my processor if I decide to add or remove something. Caching in another repo might not be wise if a new plugin comes out or a new version of one, you would be putting your users behind a dependency wall.

**Comment by bendyorke** [comment]
@stephenway yeah, but on the other hand it means no node environment is required.  By including them in the build, you can guarantee that they work, plus polyfill any node dependancies.  Likely I'll just make a shell script that updates them and runs tests, so they shouldn't ever be too far behind.

Another feature for v1 is allowing you to opt in to use a node env.  This would make life a whole lot easier and you could just specify which packages and versions you'd like to use :)

I'm trying to build with progressive enhancement in mind.  If someone install Java, they should be able to use it effectively without having external dependancies.
